integrationTests/chainSimulator/staking/jail_test.go

```diff
@@ -0,0 +1,145 @@
+package staking
+
+import (
+	"encoding/hex"
+	"fmt"
+	"math/big"
+	"testing"
+	"time"
+
+	"github.com/multiversx/mx-chain-core-go/core"
+	"github.com/multiversx/mx-chain-core-go/data/transaction"
+	"github.com/multiversx/mx-chain-go/config"
+	"github.com/multiversx/mx-chain-go/node/chainSimulator"
+	"github.com/multiversx/mx-chain-go/node/chainSimulator/components/api"
+	"github.com/multiversx/mx-chain-go/node/chainSimulator/configs"
+	"github.com/multiversx/mx-chain-go/vm"
+	"github.com/stretchr/testify/require"
+)
+
+// Test description
+// All test cases will do a stake transaction and wait till the new node is jailed
+// testcase1 -- unJail transaction will be sent when staking v3.5 is still action --> node status should be `new` after unjail
```

Comment:
I think the expected behavior for first scenario before stakingV4 here is that after the node is moved to new, it will go to waiting, then eligibile. 
Same for other as well, after auction they should probably be sent to auction -> waiting -> eligible (if enough slots).

We could fast forward more epochs and pass a []string list of expected behaviors for multiple epochs. No strong opinion about these changes, though, these are nice to have

Reply:
done